packages/cli/src/lib/commands/space.ts

```diff
@@ -148,17 +148,33 @@ export const space: CommandDefinition = (program) => {
 
     accessKeyCmd.command("revoke")
         .description("Revokes Access Key in active Space")
-        .argument("<access_key>", "Access Key id")
-        .action(async (accessKey: string) => {
+        .option("--id <id>", "Hub Id")
+        .option("--all", "Disconnects all self-hosted Hubs connected to Space")
+        .action(async (options: { all: boolean, id: string }) => {
             const spaceName = sessionConfig.lastSpaceId;
             const mwClient = getMiddlewareClient();
 
             if (!spaceName) {
                 throw new Error("No Space set");
             }
+            if (options.all) {
+                const apiKeys = await mwClient.listAccessKeys(spaceName);
+
+                if (!apiKeys.accessKeys || apiKeys.accessKeys.length === 0) {
+                    throw new Error("There are no keys to revoke");
+                } else {
+                    const revokedAccessKeys = await mwClient.revokeAllAccessKeys(spaceName, apiKeys);
 
-            const revokedAccessKey = await mwClient.revokeAccessKey(spaceName, accessKey);
+                    displayObject(revokedAccessKeys, profileManager.getProfileConfig().format);
 
-            displayObject(revokedAccessKey, profileManager.getProfileConfig().format);
+                    process.exit(1);
```

Comment:
worst possible, end... it skips all response checks etc if ever added to flow. Simple `return` does the work